src/metrics/handle.rs

```diff
@@ -2,6 +2,43 @@ use metrics::GaugeValue;
 use std::slice;
 use std::sync::atomic::{AtomicU32, AtomicU64, Ordering};
 
+#[derive(Debug)]
+struct AtomicF64 {
+    inner: AtomicU64,
+}
+
+impl AtomicF64 {
+    fn new(init: f64) -> Self {
+        Self {
+            inner: AtomicU64::new(init.to_bits()),
+        }
+    }
+
+    pub fn fetch_update<F>(
```

Comment:
Since `AtomicF64` isn't actually exposed anywhere (and the type is private), making these methods public should be superfluous, and may inhibit inlining what are otherwise fairly trivial functions.